beastify/content_scripts/beastify.js

```diff
@@ -1,37 +1,48 @@
-/*
-beastify():
-* removes every node in the document.body,
-* then inserts the chosen beast
-* then removes itself as a listener 
-*/
-function beastify(request, sender, sendResponse) {
-  removeEverything();
-  insertBeast(request.beastURL);
-  browser.runtime.onMessage.removeListener(beastify);
-}
+(function() {
+  /*
+  Check and set a global guard variable.
+  If this content script is injected into the same page again,
+  it will do nothing next time.
+  */
+  if (window.hasRun) {
+    return;
+  }
+  window.hasRun = true;
+
+  /*
+  beastify():
+  * removes every node in the document.body,
+  * then inserts the chosen beast
+  */
+  function beastify(request, sender, sendResponse) {
+    removeEverything();
+    insertBeast(request.beastURL);
+  }
+
+  /*
+  Remove every node under document.body
+  */
+  function removeEverything() {
+    while (document.body.firstChild) {
+      document.body.firstChild.remove();
+    }
+  }
 
-/*
-Remove every node under document.body
-*/
-function removeEverything() {
-  while (document.body.firstChild) {
-    document.body.firstChild.remove();
+  /*
+  Given a URL to a beast image, create and style an IMG node pointing to
+  that image, then insert the node into the document.
+  */
+  function insertBeast(beastURL) {
+    var beastImage = document.createElement("img");
+    beastImage.setAttribute("src", beastURL);
+    beastImage.setAttribute("style", "width: 100vw");
+    beastImage.setAttribute("style", "height: 100vh");
```

Comment:
These two lines don't work, the last line overwrites the previous one.

Try

```javascript
beastImage.style.width = "100vw";
beastImage.style.height = "100vh";
```

or

```javascript
beastImage.style.cssText = "width: 100vw; height: 100vh;";
```